File: btgerror.jl
"""
  The Error file
"""
module BtgError

export set_msg, set_status, get_status, get_msg

error_status = 0
error_string = ""

function set_msg(s)
    """
    Set the error message [s] to [error_string]
    """
    error_string = s
end

function set_status()
    """
    Set [error_status] to 0
    """
    error_status = 0
end

function get_status()
    """
    Get [error_status]
    """
    return error_status
end

function get_msg()
    """
    Get [error_string]
    """
    return error_string
end

end

File: btg013/src/btgerror.cc
//==========================================================================
// btg - Bayesian Trans-Gaussian Prediction program
//==========================================================================

// btgerror.cc

// Written by David Bindel

// btg program error handling code

#include <string.h>
#include <iostream.h>
#include "btgerror.hh"

//==========================================================================

// Error message buffer and status flag

char BtgError::error_string[256];
int BtgError::error_status = 0;

// Set an error
void BtgError::Set(char *s)
{
  strcpy(error_string, s);
  error_status = 1;
}

// Clear error
void BtgError::Clear()
{
  error_status = 0;
}

// Return error status flag
int BtgError::Status()
{
  return error_status;
}

// Return error message
char *BtgError::Message()
{
  return error_string;
}
